### ISSUE TITLE
Memory leak in Bus.js from permanently keeping messageListeners

### ISSUE THREAD

**Issue by vaughnkoch** [iddan/react-native-canvas#225]
Hi, thanks for this library. I'm calling fillRect(...) 10-100 times per second, which exposed a pretty big memory leak. I traced it to Bus.js in this library and I think I found the root cause.

Bus.js:
```
  /**
   * @param {Message} message 
   * @return {Promise.<Message>}
   */
  post(message) {
    return new Promise(resolve => {
      this._messageListeners[message.id] = resolve;
      if (!this._paused) {
        this._send(message);
      } else {
        this._queue.push(message);
      }
    });
  }
  /**
   * @param {Message} message 
   * @return {void}
   */
  handle(message) {
    const handler = this._messageListeners[message.id];
    if (handler) {
      handler(message);
    }
    else {
      console.warn('Received unexpected message', message);
    }
  }
```

In this code, `this._messageListeners` gets a new entry (resolve, in the code above) for each message, but those entries are never removed.

I added this code to test removal:
```
  handle(message) {
    const handler = this._messageListeners[message.id];
    console.log('before delete', Object.keys(this._messageListeners).length)
    delete this._messageListeners[message.id];
    console.log('after delete', Object.keys(this._messageListeners).length)
    ...
```
This greatly reduces the amount of memory used since it's deleting the messages as they're handled. However, there's still a slow leak since some of the messages are never handled, so the entries in `this._messageListeners` are never deleted. I didn't debug any further.

Sample output:
```
 LOG  before delete 1242
 LOG  after delete 1241
 LOG  before delete 1247
 LOG  after delete 1246
 LOG  before delete 1246
 LOG  after delete 1245
 LOG  before delete 1245
 LOG  after delete 1244
```

**Comment by iddan** [comment]
As soon as merged. I’ll give it a look either today or tomorrow

**Comment by iddan** [comment]
Published in react-native-canvas@0.1.38
